packages/preview-middleware-client/src/cpe/rta-service.ts

```diff
@@ -57,6 +58,9 @@ export class RtaService {
             // eslint-disable-next-line fiori-custom/sap-no-location-reload
             location.reload();
         });
+        this.rta.attachStart(() => {
+            sendAction(appLoaded());
```

Comment:
I suppose now this was the last thing to load, but I'm not convinced that this will always be reliable. We also need other services to have finished being initialised. We probably should just return promise when initialising this service which is fulfilled when `start` event is fired.

Reply:
adapted